src/fromager/commands/build.py

```diff
@@ -170,8 +172,20 @@ def _is_wheel_built(
 
     try:
         logger.info(f"{req.name}: checking if {req} was already built")
-        sdist.resolve_prebuilt_wheel(wkctx, req, [wkctx.wheel_server_url])
-        return True
+        url, _ = sdist.resolve_prebuilt_wheel(wkctx, req, [wkctx.wheel_server_url])
+        build_tag_from_settings = wkctx.settings.build_tag(req.name, dist_version)
+        build_tag = build_tag_from_settings if build_tag_from_settings else (0, "")
+        wheel_filename = urlparse(url).path.rsplit("/", 1)[-1]
+        _, _, build_tag_from_name, _ = parse_wheel_filename(wheel_filename)
+        existing_build_tag = build_tag_from_name if build_tag_from_name else (0, "")
+        if (
+            existing_build_tag[0] > build_tag[0]
+            and existing_build_tag[1] == existing_build_tag[1]
```

Comment:
This condition looks weird. Isn't the second part comparing the same values?

I remember we wanted the `existing_build_tag[0] > build_tag[0]` part. I'm not sure what the second part is meant to be checking.

Reply:
I think at some point @tiran wants to add variants to the build tag so it would end up looking like: `0.cpu`. So it only makes sense to check the build number if the variant matches. Right now it is a no-op but is immediately usable if we whenever we decide to introduce variants to build tags

Reply:
OK, but it's literally comparing a variable to itself so it's always going to be true. Is it meant to be comparing the same part of 2 different variables?

Reply:
ahhh i completely missed that. Thank you!